### PR TITLE
optim: validate ns_steps and eps in Muon.__init__

### PR DESCRIPTION
**Pull request by Dev-X25874**
## Description

`Muon.__init__` validates `lr`, `momentum`, and `weight_decay` eagerly, but
`ns_steps` and `eps` were never validated. This causes three silent failure
modes:

- **`ns_steps < 1`** (e.g. `ns_steps=0`): silently accepted; the
  Newton-Schulz loop runs zero iterations and returns a bare normalized
  gradient instead of an orthogonalized update — wrong behavior with no
  error raised.
- **`ns_steps >= 100`**: silently accepted at construction, then raises a
  `ValueError` deep inside `_zeropower_via_newtonschulz` on the first
  `step()` call, producing a confusing delayed traceback.
- **`eps <= 0`**: silently accepted; a non-positive `eps` defeats the
  `clamp(min=eps)` guard in `_zeropower_via_newtonschulz`, allowing
  division by zero for zero-gradient tensors.

## Changes

- `torch/optim/_muon.py`: add two `ValueError` checks in `__init__` —
  `1 <= ns_steps < 100` and `eps > 0` — consistent with the existing
  validation style for other hyperparameters in the same method.
- `torch/testing/_internal/common_optimizers.py`: add three new
  `ErrorOptimizerInput` entries to `optim_error_inputs_func_muon` covering
  `ns_steps=0`, `ns_steps=100`, and `eps=-1.0`, all as
  `CONSTRUCTION_ERROR`. These entries fail on the original code and pass
  after the fix.

## Motivation and Context

Eager validation at construction is the established pattern for all
PyTorch optimizers. Moving the `ns_steps >= 100` guard from the runtime
path into `__init__` makes the API consistent and gives users a clear error
message at the point of misconfiguration rather than mid-training.

## How Has This Been Tested?

- Three new `ErrorOptimizerInput` entries in
  `optim_error_inputs_func_muon` (in `common_optimizers.py`), exercised
  by the existing `test_errors` parametrized test in `test/optim/test_optim.py`.
- Manually verified that `Muon(params, ns_steps=0)`,
  `Muon(params, ns_steps=100)`, and `Muon(params, eps=-1.0)` all raise
  `ValueError` after the fix and pass cleanly with valid arguments.

## Checklist
- [x] No BC-breaking change (only tightens validation on previously
  undefined/erroneous inputs)
- [x] New tests added and follow existing test file conventions
- [x] `mypy` type annotations unchanged

### CODE REVIEW

File: torch/optim/_muon.py [hunk 1]
```diff
@@ -105,6 +105,12 @@ def __init__(
             raise ValueError(f"momentum should be >= 0 but is: {momentum}")
         if not 0.0 <= weight_decay:
             raise ValueError(f"weight decay should be >= 0 but is: {weight_decay}")
+        if not 1 <= ns_steps < 100:
```
Comment (janeyx99) [inline torch/optim/_muon.py:108]: I'm not sure this is the right place to error. If so then we should remove the internal error. This is a design question that's slightly less trivial than the rest of the PR

**Review by janeyx99 — CHANGES_REQUESTED** [review]
